apps/studio/components/interfaces/Realtime/RealtimeSettings.tsx

```diff
@@ -0,0 +1,345 @@
+    // [Joshen] Temp typed with any - API typing marks all the properties as nullable,
+    // but checked with Filipe that they're not supposed to
+    if (data) form.reset(data as any)
+  }, [isSuccess])
+
+  return (
+    <ScaffoldSection isFullWidth>
+      <Form_Shadcn_ {...form}>
+        <form id={formId} onSubmit={form.handleSubmit(onSubmit)}>
+          {isError ? (
+            <AlertError error={error} subject="Failed to retrieve realtime settings" />
+          ) : (
+            <Card>
+              <CardContent>
+                <FormField_Shadcn_
+                  control={form.control}
+                  name="private_only"
+                  render={({ field }) => (
+                    <FormSection
+                      className="!p-0 !pt-2"
+                      header={<FormSectionLabel>Channel restrictions</FormSectionLabel>}
+                    >
+                      <FormSectionContent loading={isLoading}>
+                        <FormItemLayout
+                          layout="flex"
+                          label="Private channels only"
+                          description="If this is enabled, only private channels will be allowed"
+                        >
+                          <FormControl_Shadcn_>
+                            <Switch
+                              checked={field.value}
+                              onCheckedChange={field.onChange}
+                              disabled={!canUpdateConfig}
+                            />
+                          </FormControl_Shadcn_>
+                        </FormItemLayout>
+                      </FormSectionContent>
+                    </FormSection>
+                  )}
+                />
+              </CardContent>
+              <CardContent>
+                <FormField_Shadcn_
+                  control={form.control}
+                  name="connection_pool"
+                  render={({ field }) => (
+                    <FormSection
+                      className="!p-0 !py-2"
+                      header={
+                        <FormSectionLabel
+                          description={
+                            <p className="text-foreground-lighter text-sm !mt-1">
+                              Sets connection pool size for Realtime Authorization
+                            </p>
+                          }
+                        >
+                          Connection pool size
+                        </FormSectionLabel>
+                      }
+                    >
+                      <FormSectionContent loading={isLoading}>
+                        <FormControl_Shadcn_>
+                          <Input_Shadcn_
+                            {...field}
+                            type="number"
+                            disabled={!canUpdateConfig}
+                            value={field.value || ''}
+                            {...form.register('connection_pool', { valueAsNumber: true })}
```

Comment:
Are these `register` calls really needed?

Reply:
yeah this one's a weird one that i've been observing with react hook form
so without this (given that we've correctly declared the form schema as `xxx: z.number()`, and assigned type `number` to the input field)

if we clear the input, then type in a number again, the value somehow gets saved in the form state as a string type thereafter
![image](https://github.com/user-attachments/assets/f385acc3-068f-41e9-a605-50298c053c19)

which causes the zod validation to fail when trying to save the form - so this register enforces that the input value _always_ gets parses as a number

although do let me know if there's a better way though 🙏